### ISSUE TITLE
Accurate in validation but got wrong result on new test data

### ISSUE THREAD

**Issue by Barry-Liang** [Microsoft/singleshotpose#153]
I have generated a custom dataset using the object data tools mentioned in this repository. After validation, the final result is as follow:
![Screenshot from 2021-01-27 20-52-38](https://user-images.githubusercontent.com/43359812/105988142-94c8e000-60e2-11eb-8565-ec9064fd92ae.png)

![Screenshot from 2021-01-27 20-45-14](https://user-images.githubusercontent.com/43359812/105987162-394a2280-60e1-11eb-9d09-736e543854b3.png)
I think this result is very good. However, I want to use it in a real-time detection case. When I use the same camera and object to detect new data, the test results are totally wrong.
![Screenshot from 2021-01-27 20-54-20](https://user-images.githubusercontent.com/43359812/105988166-9e524800-60e2-11eb-8577-c10b5db16886.png)

![Screenshot from 2021-01-27 20-54-46](https://user-images.githubusercontent.com/43359812/105988172-a14d3880-60e2-11eb-9f2c-2b3c83d256bc.png)

![Screenshot from 2021-01-27 20-56-27](https://user-images.githubusercontent.com/43359812/105988190-a8744680-60e2-11eb-8143-89e9f12c0962.png)

![Screenshot from 2021-01-27 20-58-18](https://user-images.githubusercontent.com/43359812/105988211-add19100-60e2-11eb-937a-ec3734e6b984.png)

I am sure that the program I used to test the new data is correct.  Because When I use it to test the original dataset, the results are all correct. But for the new data, the results are all wrong.

Do you have any idea about what happened? 
I guess since the video is captured continuously, all the images in the test dataset have similar images in the train set, therefore the final result is still overfitted. But I am not sure. 
I will appreciate it if you can help me with that.

**Comment by ValiaVl** [comment]
> 

Hi @Barry-Liang . Can you share the code for visualizing the results on new unseen images without the ground truth labels required?
